common/cuda_hip/matrix/hybrid_kernels.hpp.inc

```diff
@@ -140,3 +140,63 @@ __global__ __launch_bounds__(default_block_size) void add(
 
 
 }  // namespace kernel
+
+
+template <typename ValueType, typename IndexType>
+struct hybrid_tuple_unpack_functor {
```

Comment:
This is only necessary because a Lambda wouldn't be copy-assignable, which one of the algorithms requires. So this is only a local workaround without any generic usage.